nixGL.nix

```diff
@@ -127,36 +153,17 @@ let
       nixVulkanNvidia = nixNvidiaWrapper "Vulkan";
     };
 
-    nixGLIntel = writeExecutable {
-      name = "nixGLIntel";
-      # add the 32 bits drivers if needed
-      text = let
-        mesa-drivers = [ mesa.drivers ]
-          ++ lib.optional enable32bits pkgsi686Linux.mesa.drivers;
-        intel-driver = [ intel-media-driver vaapiIntel ]
-          # Note: intel-media-driver is disabled for i686 until https://github.com/NixOS/nixpkgs/issues/140471 is fixed
-          ++ lib.optionals enable32bits [ /* pkgsi686Linux.intel-media-driver */ driversi686Linux.vaapiIntel ];
-        libvdpau = [ libvdpau-va-gl ]
-          ++ lib.optional enable32bits pkgsi686Linux.libvdpau-va-gl;
-        glxindirect = runCommand "mesa_glxindirect" { } (''
-          mkdir -p $out/lib
-          ln -s ${mesa.drivers}/lib/libGLX_mesa.so.0 $out/lib/libGLX_indirect.so.0
-        '');
-      in ''
-        #!${runtimeShell}
-        export LIBGL_DRIVERS_PATH=${lib.makeSearchPathOutput "lib" "lib/dri" mesa-drivers}
-        export LIBVA_DRIVERS_PATH=${lib.makeSearchPathOutput "out" "lib/dri" intel-driver}
-        ${''export __EGL_VENDOR_LIBRARY_FILENAMES=${mesa.drivers}/share/glvnd/egl_vendor.d/50_mesa.json${
-          lib.optionalString enable32bits
-          ":${pkgsi686Linux.mesa.drivers}/share/glvnd/egl_vendor.d/50_mesa.json"
-          }"''${__EGL_VENDOR_LIBRARY_FILENAMES:+:$__EGL_VENDOR_LIBRARY_FILENAMES}"''
-        }
-        export LD_LIBRARY_PATH=${lib.makeLibraryPath mesa-drivers}:${lib.makeSearchPathOutput "lib" "lib/vdpau" libvdpau}:${glxindirect}/lib:${lib.makeLibraryPath [libglvnd]}"''${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
-        exec "$@"
-      '';
-    };
 
-    nixVulkanIntel = writeExecutable {
+    nixGLMesa = writeNixGL "nixGLMesa" [  ];
+
+    nixGLMesaVdpau = writeNixGL "nixGLMesaVdpau" [ vaapiVdpau ];
+
+    nixGLIntel = writeNixGL "nixGLIntel"
+      ([ intel-media-driver vaapiIntel ]
+       # Note: intel-media-driver is disabled for i686 until https://github.com/NixOS/nixpkgs/issues/140471 is fixed
```

Comment:
That linked issue got closed over a year ago.

Reply:
This is from the original implementation, I can of course delete the comment it. However I do not have an intel system, so I don't feel like changing the Intel stuff too much.

Reply:
I have an intel system and maintain intel-media-driver. This can be removed for 22.11 and unstable.

Reply:
Great, I removed it!